Completed Lesson-7

borisshavrin opened the pull request.

borisshavrin committed:
Completed Lesson-7

pantonov-gb commented on task1.py:
```diff
@@ -0,0 +1,32 @@
+
+
+class Matrix:
+    def __init__(self, row, column):            # конструктор класса, принимающий кол-во строк и колонок матрицы
```
> обеспечить перегрузку конструктора класса (метод __init__()), который должен принимать данные (список списков) для формирования матрицы.

А тут происходит генерация списка списков

pantonov-gb commented on task1.py:
```diff
@@ -0,0 +1,32 @@
+        self.matrix = [[random.randrange(10) for j in range(column)] for i in range(row)]   # генерир-е матрицы
+
+    def __str__(self):                          # метод вывода матрицы
+        return f'{self.matrix}'
```
В выводе всего одна строка, независимо от кол-ва строк в матрице. Привычно было бы видеть вывод матрицы - по строке на экране на каждую строку в матрице, например.

pantonov-gb commented on task2.py:
```diff
@@ -0,0 +1,41 @@
+    def __init__(self, param):
+        self.param = param
+
+    @classmethod                    # создание абстрактного метода
```
classmethod делает метод не абстрактным, а "методом класса".

Декоратор для абстрактного метода это `abc.abstractmethod`

pantonov-gb commented on task2.py:
```diff
@@ -0,0 +1,41 @@
+
+class Clothes(ABC):
+    def __init__(self, param):
+        self.param = param
```
Непонятно, что это за param. Нужно выбрать более подходящее названия для передачи роста и размера. При этом необходимо учесть, что рост нужен только для костюма, а размер - только для пальто. Другими словами, должно быть два разных конструктора с разным набором параметров.

pantonov-gb commented on task3.py:
```diff
@@ -0,0 +1,71 @@
+
+    def __add__(self, other):
+        sum_qty = self.qty + other.qty
+        sum_qqq = self.qqq + other.qqq
```
Что такое qqq?

pantonov-gb commented on task3.py:
```diff
@@ -0,0 +1,71 @@
+
+        return f'Суммарное кол-во клеток: {sum_qty}, Число ячеек общей клетки: {sum_qqq}'
+
+    def __sub__(self, other):
```
Данный метод, а также другие (`__add__`, `__mul__`) - должны возвращать объект типа Cell

pantonov-gb reviewed (commented):
Нужно доработать

borisshavrin committed:
modified task1

1. Убрал генерацию матриц в конструкторе. 
Теперь на ввод принимается сгенерированная отдельно матрица.

2. Исправил метод str. Теперь матрица выводится построчно и с пробелами.

borisshavrin committed:
modified task2

1. Исправил декоратор @classmethod на @abstractmethod
2. Изменил конструктор для передачи параметров роста и размера

borisshavrin committed:
modified task3

1. Убрал переменную qqq
2. Исправил методы, чтобы они возвращали объекты типа Cell
3. Метод make_order(self, qqq) - вместо qqq принимает переменную 
row - количество ячеек в ряду

pantonov-gb reviewed (approved):
Ok
